sys/targets/targets.go

```diff
@@ -863,7 +855,7 @@ func (target *Target) replaceSourceDir(param *string, sourceDir string) {
 }
 
 func (target *Target) lazyInit() {
-	if runtime.GOOS != target.BuildOS || target.BrokenCompiler != "" {
+	if (target.OS != TestOS && runtime.GOOS != target.BuildOS) || target.BrokenCompiler != "" {
```

Comment:
For TestOS we should set BuildOS = runtime.GOOS in init func. Is this needed?

Reply:
Hmm, this
```
# go test -v -count=1 -run=TestCover/32/11 ./pkg/runtest/
=== RUN   TestCover
=== PAUSE TestCover
=== CONT  TestCover
lazyInit: GOOS=linux OS=test BuildOS=test
=== RUN   TestCover/32
=== PAUSE TestCover/32
lazyInit: GOOS=linux OS=test BuildOS=linux
=== CONT  TestCover/32
    build.go:39: failed to build executor: failed to build program:

        executor/executor.cc:7:10: fatal error: pkg/flatrpc/flatrpc.h: No such file or directory
            7 | #include "pkg/flatrpc/flatrpc.h"
              |          ^~~~~~~~~~~~~~~~~~~~~~~
        compilation terminated.

        compiler invocation: g++ [-o /tmp/syz-executor1053807441 -DGOOS_test=1 -DGOARCH_32=1 -DHOSTGOOS_linux=1 executor/executor.cc -Wno-overflow]
--- FAIL: TestCover (0.24s)
    --- FAIL: TestCover/32 (0.01s)
FAIL
FAIL    github.com/google/syzkaller/pkg/runtest 0.365s
FAIL
```

Reply:
Is BuildOS = "test" valid at all ?

Reply:
I think it's set in `initTarget()`:

```
	if target.BuildOS == "" {
		target.BuildOS = OS
	}

```

Reply:
this would also make the commit with spoiling compiler fix unnecessary :partying_face:

Reply:
> Is BuildOS = "test" valid at all ?

No, it's not. We use BuildOS to test if we can build for this target on the current machine (i.e. it's compared with runtime.GOOS).

But for "test" OS we set BuildOS to runtime.GOOS here:
https://github.com/google/syzkaller/blob/master/sys/targets/targets.go#L657

Reply:
I think this is responsible:
```
		host := List[goos][arch]
		if host == nil {
                        fmt.Printf("Arch=%v\n", target.Arch)
			continue
		}

```

```
Arch=32
Arch=32_fork
```